BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationManager.m

```diff
@@ -602,11 +606,19 @@ -(void)inAppDidDismiss:(NSDictionary *)notificationPayload fromViewController:(B
 
 -(void)inAppActionDidTapped:(NSDictionary *)notificationPayload fromViewController:(BlueShiftNotificationViewController *)controller {
     [[BlueShift sharedInstance] trackInAppNotificationButtonTappedWithParameter: notificationPayload canBacthThisEvent: NO];
+    // invoke the inApp clicked callback method
+    if ([[[BlueShift sharedInstance].config inAppNotificationDelegate] respondsToSelector:@selector(inAppNotificationDidClick:)]) {
+        [[[BlueShift sharedInstance].config inAppNotificationDelegate] inAppNotificationDidClick:notificationPayload];
```

Comment:
`notificationPayload` will contain the `clk_elmt` value right? I did not see any special code here for that here. Hope it is already taken care of.

Reply:
yes, it will have the `clk_elmt` and `clk_url`. This was implemented when we added clk_elmt and clk_url for tracking, so its already there.